src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6426,16 +6426,126 @@ internal Ast CommandRule(bool forDynamicKeyword)
 
         #region Expressions
 
-        private ExpressionAst ExpressionRule()
+        private ExpressionAst ExpressionRule(bool endNumbeOnTernaryOpChars = false)
         {
             // G  expression:
             // G      logical-expression
             // G
             // G  logical-expression:
+            // G      binary-expression
+            // G      ternary-expression
+            // G
+            // G  ternary-expression:
+            // G      binary-expression  new-lines:opt   '?'   new-lines:opt   ternary-expression   new-lines:opt   ':'   new-lines:opt   ternary-expression
+
+            RuntimeHelpers.EnsureSufficientExecutionStack();
+            var oldTokenizerMode = _tokenizer.Mode;
+            try
+            {
+                SetTokenizerMode(TokenizerMode.Expression);
+
+                List<Ast> componentAsts = new List<Ast>();
+                ExpressionAst condition = BinaryExpressionRule(endNumbeOnTernaryOpChars);
```

Comment:
`endNumbeOnTernaryOpChars` --> endNumbe**r**OnTernaryOpChars, was this a typo ?

Reply:
Yeah, a typo :) Thanks for catching it.

Reply:
Fixed.